### PR TITLE
C Sharp Program to calculate the percentage for an exam of a student.

### PR DESCRIPTION
**Pull request by swayam1411**
Use IResult interface.Implement for SSC,NIIT,IIT exams.

### CODE REVIEW

File: Calculate_Result/Calculate_Result/AllExam.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                Console.Write($"Enter Marks Of {i + 1} Subject outof(100) => ");
+                subjectArr[i] = Convert.ToInt32(Console.ReadLine());
+            }
+            int total = subjectArr.Sum();
+            double percentage = total / 5;
+            return percentage;
+        }
+    }
+
+    public class NiitResult : IResultInterface
```
Comment (mmeshram) [inline Calculate_Result/Calculate_Result/AllExam.cs:23]: Can you rename NiitResult as NIITResult?

Comment (mmeshram) [inline Calculate_Result/Calculate_Result/AllExam.cs]: `public class NiitResult : IResultInterface`
Can you rename NiitResult by NIITResult same for SSC?

Comment (swayam1411) [reply Calculate_Result/Calculate_Result/AllExam.cs]: okay can i do same for Ssc & iit ?

File: Calculate_Result/Calculate_Result/AllExam.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        public double CalculateResult()
+        {
+            Console.Write("Enter CGPA Of NIIT outof(10) => ");
+            double cgpa = Convert.ToDouble(Console.ReadLine());
+            double percentage = cgpa * 9.5;
+            return percentage;
+        }
+    }
+
+    public class IitResult : IResultInterface
```
Comment (mmeshram) [inline Calculate_Result/Calculate_Result/AllExam.cs]: public class IitResult : IResultInterface
Rename IitResult by IITResult.

File: Calculate_Result/Calculate_Result/MainClass.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Calculate_Result
+{
+    class MainClass
+    {
+        public static void AddStudent(List<Student> Students)
+        {
+            IResultInterface IObject;
```
Comment (mmeshram) [inline Calculate_Result/Calculate_Result/MainClass.cs]: ` IResultInterface IObject`
Make reference variable as camelCase i.e iObject

File: Calculate_Result/Calculate_Result/AllExam.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    }
+
+    public class NIITResult : IResultInterface
+    {
+        public double CalculateResult()
+        {
+            Console.Write("Enter CGPA Of NIIT outof(10) => ");
+            double cgpa = Convert.ToDouble(Console.ReadLine());
+            double percentage = cgpa * 9.5;
```
Comment (mmeshram) [inline Calculate_Result/Calculate_Result/AllExam.cs]: `double percentage = cgpa * 9.5;
            return percentage;`
Can we do return cgpa * 9.5 instead?

File: Calculate_Result/Calculate_Result/AllExam.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        public double CalculateResult()
+        {
+            int[] subjectArr = new int[5];
+
+            for (int i = 0; i < subjectArr.Length; i++)
+            {
+                Console.Write($"Enter Marks Of {i + 1} Subject outof(100) => ");
+                subjectArr[i] = Convert.ToInt32(Console.ReadLine());
+            }
+            int total = subjectArr.Sum();
```
Comment (mmeshram) [inline Calculate_Result/Calculate_Result/AllExam.cs]: Can we factor bellow code ?
`   int total = subjectArr.Sum();
        double percentage = total / 5;
         return percentage;`
as return subjectArr.Sum()/5; ?

**Review by mmeshram — APPROVED** [review]

**Pull request merged**